src/main/java/Duke.java

```diff
@@ -1,10 +1,120 @@
+import java.util.*;
 public class Duke {
+    private static ArrayList<Task> lists = new ArrayList<>();
     public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        String inputs;
+        String[] splitInputs;
+        doGreetings();
+        inputs = scanner.nextLine();
+        while(!(inputs.equalsIgnoreCase("bye"))) {
+            System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+            splitInputs = inputs.split(" ");
+            if(inputs.equals("list")){
+                doPrintlists();
+            }
+            else if(splitInputs[0].equalsIgnoreCase("mark")){
+                if(splitInputs.length == 1){
+                    System.out.println("No index given to mark");
+                }
+                else if(Integer.parseInt(splitInputs[1]) > lists.size()){
+                    System.out.println("There is no task with index " + splitInputs[1]);
+                }
+                else {
+                    lists.get(Integer.parseInt(splitInputs[1]) - 1).setDone(true);
+                    doPrintlists();
+                }
+            }
+            else if(splitInputs[0].equalsIgnoreCase("unmark")){
+                if(splitInputs.length == 1){
+                    System.out.println("No index given to unmark");
+                }
+                else if(Integer.parseInt(splitInputs[1]) > lists.size()){
+                    System.out.println("There is no task with index " + splitInputs[1]);
+                }
+                else {
+                    lists.get(Integer.parseInt(splitInputs[1]) - 1).setDone(false);
+                    doPrintlists();
+                }
+            }
+            else if(splitInputs[0].equalsIgnoreCase("todo") || splitInputs[0].equalsIgnoreCase("event") || splitInputs[0].equalsIgnoreCase("deadline")){
+                lists.add(new Task(splitInputs));
+                System.out.println("\t Added: ");
+                lists.get(lists.size() - 1).doPrint();
+                System.out.println("\t There are currently " + lists.size() + " task(s) in the list :)");
+            }
+            else {
+                System.out.println("Sorry I do not understand your command :(");
+            }
```

Comment:
This code block is hard to follow. Do consider applying the Single Level of Abstraction Principle (SLAP)